src/com/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1264,3 +1243,21 @@
   [action-for-unsupported-fn]
   (fail-on-unsupported action-for-unsupported-fn)
   (warn-on-db-deprecation))
+
+(def ^:dynamic *orphaned-path-gc-limit* 200)
+(def ^:dynamic *orphaned-value-gc-limit* 200)
+
+(defn garbage-collect!
+  "Delete any lingering, unassociated data in the database"
+  [db]
+  (time!
+   (:gc metrics)
+   (jdbc/with-transacted-connection db
```

Comment:
@senior I think you're right about the semantics, but want to make sure it's for the same reason.  Previously, the with-transacted-connection here was in the caller (garbage-collect! in services.clj).  I just pushed it down into storage/garbage-collect! so that we could put the original functions inside the same transaction they were in before, and add a new serialized transaction for the facts/paths/ gc.

Though that *is* a change to the semantics of storage/garbage-collect!; it didn't establish any of its own transactions before.  However, storage/garbage-collect! isn't called by anyone other than services.clj (aside from tests -- though those would indeed see different behavior now).  Thoughts?  I could break this up into two functions, so that we can leave the transaction wrappers in services.clj, but I wondered if we really wanted servcies.clj to have to know about the differing transactional requirements.